dashboard/app/controllers/omniauth_callbacks_controller.rb

```diff
@@ -151,4 +205,14 @@ def allows_silent_takeover(oauth_user)
     lookup_user = User.find_by_email_or_hashed_email(oauth_user.email)
     allow_takeover && lookup_user
   end
+
+  def should_connect_provider?
+    return current_user && session[:connect_provider].present?
+  end
+
+  def can_connect_provider?
```

Comment:
Should `can_connect_provider?` also check for a current user?

Reply:
good call, i'll add that check